### PR TITLE
Create patchinfo factory

### PR DESCRIPTION
**Pull request by eduardoj**
~Depends on #14343.~

### CODE REVIEW

File: src/api/spec/factories/patchinfo.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :patchinfo, class: 'Patchinfo' do
+    # Patchinfo is an ActiveModel::Model. Override FactoryBot `create` method
+    to_create { nil }
```
Comment (krauselukas) [inline src/api/spec/factories/patchinfo.rb:4]: @eduardoj maybe you can use `skip_create` here? https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#custom-methods-to-persist-objects

Comment (eduardoj) [reply src/api/spec/factories/patchinfo.rb]: @krauselukas yes. Thanks!

**Review by saraycp — COMMENTED** [review]
Most of the time you end up calling `create_patchinfo`. Why don't you simply do that internally in the factory?

Calling the factory could look like this: `create(:patchinfo, project: project, package: package, comment: comment, data: data)`

**Comment by eduardoj** [comment]
> Most of the time you end up calling `create_patchinfo`. Why don't you simply do that internally in the factory?
> 
> Calling the factory could look like this: `create(:patchinfo, project: project, package: package, comment: comment, data: data)`

Good idea! I added a commit which creates two factories. Please take a look.

**Review by krauselukas — APPROVED** [review]

**Pull request merged**